vendor/engines/bulk_email/app/controllers/bulk_email/bulk_email_controller.rb

```diff
@@ -37,16 +39,26 @@ def init_search_options
       @products = current_facility.products.active_plus_hidden.order("products.name").includes(:facility)
       @search_options = { products: @products }
       @search_fields = params.merge(facility_id: current_facility.id)
-      @search_types = search_types
-      @search_types.delete(:authorized_users) unless @products.exists?(requires_approval: true)
+      @user_types = user_types
+      @user_types.delete(:authorized_users) unless @products.exists?(requires_approval: true)
     end
 
-    def search_types
-      BulkEmailSearcher::SEARCH_TYPES.each_with_object({}) do |search_type, hash|
-        hash[search_type] = I18n.t("bulk_email.search_type.#{search_type}")
+    def datepicker_field_input(form, key)
+      date = @search_fields[key].presence.try(:tr, "-", "/") || ""
```

Comment:
Ok, I was thinking we had a standard way of dealing with this. I think it's fine.

Would `@search_fields[key].to_s.tr("-", "/")` be equivalent? It would get rid of the `try` and the `||`.